File: Data Structure & Algorithm/Archive/Data Structure/String/en/easy/557. Reverse Words in a String III/557. Reverse Words in a String III.py
# 557. Reverse Words in a String III

# Given a string s, reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order.

# Example 1:
# Input: s = "Let's take LeetCode contest"
# Output: "s'teL ekat edoCteeL tsetnoc"

# Example 2:
# Input: s = "God Ding"
# Output: "doG gniD"


class Solution:

    def reverseWords(self, s: str) -> str:
        arr = s.split()
        for i in range(len(arr)):
            arr[i] = arr[i][::-1]
        return ' '.join(arr)


# test
string = "Let's take LeetCode contest"
solution = Solution()
res = solution.reverseWords(string)
print(res)

File: Data Structure & Algorithm/Topic/Data Structure/String/en/easy/557. Reverse Words in a String III/557. Reverse Words in a String III.ts
// 557. Reverse Words in a String III

// Given a string s, reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order.

// Example 1:
// Input: s = "Let's take LeetCode contest"
// Output: "s'teL ekat edoCteeL tsetnoc"

// Example 2:
// Input: s = "God Ding"
// Output: "doG gniD"

// use two point to switch elements, and reverse() can do it too.
export function reverseWords(s: string): string {
  let arr: string[] = s.split(' ');
  arr = arr.map((value) => {
    let word = value.split('');
    let left: number = 0;
    let right: number = value.length;
    // 双指针
    while (left < right) {
      const temp = word[left];
      word[left] = word[right];
      word[right] = temp;
      left++;
      right--;
    }
    return word.join('');
  });
  return arr.join(' ');
}

// test
const res = reverseWords("Let's take LeetCode contest");
console.log(res);
